src/Lecture4.hs

```diff
@@ -257,7 +323,9 @@ the file doesn't have any products.
 -}
 
 calculateStats :: String -> String
-calculateStats = error "TODO"
+calculateStats c = case mapMaybe parseRow $ lines c of
+  [] -> "file has no products"
+  (r:rs) -> displayStats . combineRows $ fromList (r:rs)
```

Comment:
Here you can actually use the `:|` constructor of the `NonEmpty` data type instead of function `fromList` to create `NonEmpty` directly once you know the head and the tail already 👍🏻 
```suggestion
  (r:rs) -> displayStats . combineRows (r :| rs)
```